soh/soh/Enhancements/debugconsole.cpp

```diff
@@ -549,15 +515,15 @@ static bool AddHeartContainerHandler(std::shared_ptr<Ship::Console> Console, con
     if (gSaveContext.healthCapacity >= 0x140)
         return CMD_FAILED;
 
-    Health_GiveHearts(1);
+    GameInteractor::Actions::AddOrRemoveHealthContainers(1);
     return CMD_SUCCESS;
 }
 
 static bool RemoveHeartContainerHandler(std::shared_ptr<Ship::Console> Console, const std::vector<std::string>& args) {
     if ((gSaveContext.healthCapacity - 0x10) < 3)
         return CMD_FAILED;
```

Comment:
looks like there are quite a few spots in here where we're doing checks like this. i think it makes more sense for those to exist in the gameinteractor

since we used to use the console to handle the effects from crowd control, and now we're going straight to the GameInteractor, this means these checks are being skipped when fired via CC

so for this example, if `GameInteractor::Actions::AddOrRemoveHealthContainers` had the `healthCapacity` check, and returned a `GameInteractionEffectQueryResult`, we could then just check that result value and return `CMD_FAILED` or `CMD_SUCCESS` based on that

Reply:
This was one of those cases of "I don't know if I should touch all of these in this PR already because it's already quite big", but I agree it makes sense to have them all in one place, especially when moving all the checks inside the Apply() function. I'll try and tackle the ones that have been switched to using the GameInteractor.